ACMILabs/lens-reader
# #82 Onboarding Auth functionality

Resolves #82

Add authentication to onboarding lens reader to login and update authentication

### Acceptance Criteria
Replace acceptance criteria with the acceptance criteria in the ticket or check the box if none.

### Relevant design files
Add a links to the relevant design files or leave as none.
* None

### Testing instructions
Add a list of steps required to test the feature.
1. Setup a lens reader in the e__tap-reader-pi-4 fleet
2. Set the `ONBOARDING_LEDS_API` env variable to `http://<YOUR IP ADDRESS>/`
3. Set the environment variable ONBOARDING_LEDS_DATA_SUCCESS to {"success": true}
4. Set the environment variable ONBOARDING_LEDS_DATA_FAILED to {"success": false}
6. Start up a Flask server with the code below by running `export FLASK_APP=server.py` and `flask run --host=<YOUR IP ADDRESS>`
7. start your lens reader, see that it sends a http request for a token.
8. tap your lens reader, see that it sends the token in the auth header to the trigger endpoint
9. see that it correctly refreshes the token after approximately a 60 seconds (the expiry time of the token)

testing flask server code
``` python
# server.py
from pprint import pp
from flask import Flask, request
app = Flask(__name__)


@app.route('/')
def home():
    return trigger()


@app.route('/api/trigger', methods=['GET', 'POST'])
def trigger():
    pp(request.get_json())
    print(f'Auth header: {request.headers.get("Authorization")}')
    print('received')
    return {"ok": True}


@app.route('/token', methods=['POST'])
def token():
    print(request.get_json())
    print('token request received')
    return {
        'access_token': 'testToken',
        'expires_in': 60,
        'token_type': 'Bearer'
    }


```
### DoD
For requester to complete:
